### ISSUE TITLE
False positive in `no-immutable-reactive-statements`

### ISSUE THREAD

**Issue by marekdedic** [sveltejs/eslint-plugin-svelte#484]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.

### What version of ESLint are you using?

8.40.0

### What version of `eslint-plugin-svelte` are you using?

2.28.0

### What did you do?

I am implementing a setup for loading images in pages (i.e. something like calling an API endpoint to get the first 15 images, then loading the next 15 for the next page and so on...). To do this, I am using 4 variables in this way:

```svelte
<script>
  /* eslint svelte/no-immutable-reactive-statements: "error" */

  let page = 1;
  const perPage = 15;
  $: pageStart = perPage * (page - 1);
  $: pageEnd = pageStart + perPage;

  $: imagesOnThisPage = images.slice(pageStart, pageEnd);
</script>

{#each imagesOnThisPage as image}
 ... show the image
{/each}

<div
  on:click={() => {
    page += 1;
  }}
>
  Next page
</div>
```

### What did you expect to happen?

I expected no issue, as both `pageStart` and `pageEnd` **need** to be reactive, otherwise paging won't work.

### What actually happened?

I got `svelte/no-immutable-reactive-statements` error for `pageStart` (and also `pageEnd` if I make `pageStart` non-reactive)

### Link to **GitHub Repo** with Minimal Reproducible Example

I think the example should be a repro as it is, let me know if it isn't :)

**Comment by ota-meshi** [comment]
I couldn't reproduce your problem :thinking: 

https://sveltejs.github.io/eslint-plugin-svelte/playground/#eJxlkEFPhDAQhf/KpHpgWYHswQsL3LyqiR651DJZGqHdtCOaEP67U7oQE4/9Ou+9mTcLZTsUpai8cvpKTWsAihTQD9oQ+AkHwsLYTI/jF8mPATOHUpGeMPMkCUc05EtoBTpnXSsgLVoTTAYkuMoLQg2ncwDKGs8I3euNPq74vlzH3kg6Yrr9p5Cs6gxOh79zT6YLU7viuCnOMZbH9MhP/2Lee+1vWRHlfJTCZBc/bI4hoSr2Bloz3/GR/X8n6SNbOCrPc/C9/QbqMVLWFUG3BIuq01NYyJpScexnPScHqBuYA4RYzXHrZmHJWv0z/sTawkLs0IjlF1/zizY=

**Comment by marekdedic** [comment]
Whoa, this is really strange - see [this playground](https://sveltejs.github.io/eslint-plugin-svelte/playground/#eJx1Vk2P2zYQ/SuC20PSLNdIgV4269x6boD2VhcFRY1k1hSpkJTXjuH/3jeU9Wntydab4XA48+aR141yBW1eNq9Bed3EzEhb7fabGPabr3u7t1mm68b5mH2TlbYyamez0rs622+en7fKwWjJxrAd7c/hRCbSfvOlC2AoZo2sKNtln78woJwNgMh/u6O/Jfjnl+T2Z5TYbjfYf8k+pNUi+/xx6ve7LdhrWPGpX3HftttF10ACHP/+p1/cQX/Yvw463DPooOdgtKIPQ8infh/e93XbVShV5foTSXV4jCRD9u+Nt3kt9Clz9kUh4nF3/fAx233NrrdbtsX665ZX3zjQ66SsubbFi2q9Rz13V96a3TdPG98aCpuX66Yr7DZvY3RWHGQQ8dKge78+9SY0pMZyUWhPKurTzFiQB1KkhUHWJLRt2hiEayP/Tl3pjGJqDiWN8PS9RTwRjItpx5lrqX2IQsboNRIjYbSl3JM8Tp0OsTZCGRe0rUTupTpSFKGRCt8Pft9bF+d7JDiQKfsYUyPKhjznSKmtRi5II1VBGOeaqUctz2PKQeCwKe+ZSxsi2kRraVqH1aKgvK1ElFV4tKWMV0yFq0UtrW5ag64/RC3ahgSZgN6UIjdOHR8CsAe6P3R41SHEiyHReIeTRb3u02KX96NcrKy16nqOv7Sw0zl6OdYXtDWP2+i6bqPMzaQRqGkk5tWDr7VoQkHKSJ/mYelgkUjZWsU2UNgWhvzCxeX/4TQIJSLyE30Hl5GGZPpw7zoYkMhL864df0LQlV3YwwGKySnO+3AR7kTe6+IhI6YYaCi9a7EI44ap47iBzzLwdLkoOgylDBerFhaIV4X5yiHmx6XJS214CLstF9bWHq17s/e0B3IMB3hwB4dwyAQI5IuEli6YWgphvReNpxJNR8vhsSpZdw9ULPpWxdbzduncnNJasEXqU49ex+jUSSRKEJGSf9S03tM1Udf6B/WNXG3FII8pLyWNyaFvIc1XQBsaELp+f8FAJqnQkNU0kiNzQc9kbqTZal6jebUaAdaJBs5MzI2i07D7jTIzXwJmeIqcpNHgLRosjtSxJF2/3DOHnkBuulkt8XXPxnVTA6JixlAFNLdDWECZ1NziWqOIdCbVIk5vVZKVa6kUbEhEmowkY3iiSNTQUiV+kHcjznVbuFq+yR4xCbqwvx6y7izRO74dKzr3YLoSKvLjN14/JE5yRFh6MZ9hBnSZ11Tn5OeW+2Uww1DjqRPeGBDVVJYepbq5T2v/IdQB5VKxn7e5EZMAy3A2Oi/K0Ml97pwhaXmnOLcEqnWPlJiAeICUVYcBgtAuIlbG5VC5Odi9NZeg7eiVCt0MKN5JFS5RL94OUOlE2d6GNwJeNSUIREqHSc9AJxygYP17pxy1PvO03/UYFMH9NRgtvYlwqXNnBsRZp/iNBDriNWgEBSWbIRFcSEkQhgjJuf8Av6Nj7RF5qw1eA4MbZC7Rm0YAZ59qNus/v4fmpQprwxQOsnBvADGPGicu0n0+hsF4BKaklyOrIl60IqeSBWqc3iT5BZXjB78TU8i6xQGGJ1SysbAd+PYfoSBL3Lp48BpzWaCYTzkdrzvMCuzgz/2C/t0fTOPdY2ROY3n724YFOF0HZEdW9EbFqr8E531707Hz6BX4oskUCWFV18HKLtGOmdxDx1W53f4HkOCt+g==)

If I remove the `Pagination` element, the issue disappears - as it probably should. But what is really strange is that the issue also disappears if I remove the `<div>` or `lang="ts"` - what is happening here???

**Comment by ota-meshi** [comment]
After investigating, it seems that the scope manager information returned by the parser differs depending on whether `ts` is used. I don't know the specific cause yet.
